client/rest-high-level/src/main/java/org/elasticsearch/client/security/user/User.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.security.user;
+
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.Strings;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+
+/**
+ * An authenticated user
+ */
+public final class User {
+
+    private final String username;
+    private final List<String> roles;
+    private final Map<String, Object> metadata;
+    @Nullable private final String fullName;
+    @Nullable private final String email;
+
+    public User(String username, List<String> roles, Map<String, Object> metadata, @Nullable String fullName, @Nullable String email) {
+        assert username != null;
```

Comment:
we tend to use `Objects.requireNotNull(obj, "msg")` vs `assert`

Reply:
+1 to use Objects.requireNonNull in this client side class

Reply:
ACK